src/components/item-form/AccessibleAtFormFieldArray.tsx

```diff
@@ -21,6 +22,12 @@ export function AccessibleAtFormFieldArray(props: AccessibleAtFormFieldArrayProp
   const { t } = useI18n<'authenticated' | 'common'>()
   const fieldArray = useFieldArray<string | undefined>(field.name, { subscription: {} })
 
+  useEffect(() => {
+    if (field.isRequired && fieldArray.fields.length === 0) {
+      fieldArray.fields.push(undefined)
+    }
+  }, [field.isRequired, fieldArray.fields])
```

Comment:
can we avoid this effect?

since react runs effects *after* render, this means we would render the initial field array twice. usually, updating state in an effect is something to avoid if possible.

since "recommended fields" are really "initial values" in a create-item form, we could just keep the `accessibleAt: [undefined]` there? alternatively pass the fields config from `useToolFormFields` to `useToolFormRecommendedFields` and check if accessibleAt is required before pushing `[undefined]` there, but that may be overkill.